pkg/packer/reader_test.go

```diff
@@ -0,0 +1,95 @@
+package packer
+
+import (
+	"errors"
+	"github.com/golang/mock/gomock"
+	"github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/viaduct/mocks"
+	"github.com/kubeedge/viaduct/pkg/translator"
+	"io"
+	"reflect"
+	"testing"
+)
+
+
+// mockStream is mock of interface Stream.
+var mockReader *mocks.MockReader
+
+// initMocks is function to initialize mocks.
+func initMocks(t *testing.T){
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockReader =mocks.NewMockReader(mockCtrl)
+}
+
+// TestNewReader is function to test NewReader().
+func TestNewReader(t *testing.T) {
+	var reader io.Reader
+	tests := []struct {
+		name string
+		r io.Reader
+		want *Reader
+	}{
+		{
+			name:"NewReaderTest",
+			r:reader,
+			want:&Reader{reader:reader},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NewReader(tt.r); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewReader() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+// TestRead is function to test Read().
+func TestRead(t *testing.T) {
+	initMocks(t)
+	var ioreader io.Reader
+	var msg  = model.Message{Content:"message"}
+	bytesMsg, _ :=translator.NewTran().Encode(&msg)
+	header := PackageHeader{Version:0011,PayloadLen:(uint32(len(bytesMsg)))}
+	headerBuffer := make([]byte, 0)
+	header.Pack(&headerBuffer)
+	err1:=errors.New("Error")
```

Comment:
Lets not suffix names with numbers. Please change it to something that says why this error is used.